commit b59a8efbe831b50c32b7008e247f833521c7d103 (refs/changes/16/1316/1)
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2017-01-20T23:41:16-08:00 (2 years, 9 months ago)
    
    HEAD-2344 headnode setup fails to adopt cmon-agent which leads to config-agent not starting up

diff --git a/config/sapi/services/cmon-agent/service.json b/config/sapi/services/cmon-agent/service.json
new file mode 100644
index 00000000..7f9f47c6
--- /dev/null
+++ b/config/sapi/services/cmon-agent/service.json
@@ -0,0 +1,16 @@
+{
+    "name": "cmon-agent",
+    "type": "agent",
+    "params": {
+        "tags": {
+            "smartdc_role": "cmon-agent",
+            "smartdc_type": "core"
+        }
+    },
+    "metadata": {
+        "SERVICE_NAME": "cmon-agent",
+        "CMON_AGENT_LOG_LEVEL": "info"
+    },
+    "manifests": {
+    }
+}
diff --git a/scripts/headnode.sh b/scripts/headnode.sh
index a4ce47c7..196da9ef 100755
--- a/scripts/headnode.sh
+++ b/scripts/headnode.sh
@@ -749,7 +749,7 @@ function adopt_agents()
         CONFIGURABLE_AGENTS="net-agent vm-agent cn-agent \
           agents_core amon-agent amon-relay \
           config-agent firewaller smartlogin \
-          cabase cainstsvc hagfish-watcher"
+          cabase cainstsvc hagfish-watcher cmon-agent"
 
         for agent in $CONFIGURABLE_AGENTS; do
             local instance_uuid
